File: 0001-two-sum/0001-two-sum.py
from collections import defaultdict

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = defaultdict(int)

        for i, num in enumerate(nums):
            val = target - num

            if val in seen:
                return [i, seen[val]]
            
            seen[num] = i
            

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let map = new Map();

    for( let i=0; i<nums.length; i++) {
        let val = target - nums[i];

        if( map.has(val) )
            return [map.get(val), i]

        map.set(nums[i], i)
    }
};